lib/puppet/type/file.rb

```diff
@@ -51,7 +51,7 @@ def self.title_patterns
 
     validate do |value|
       unless Puppet::Util.absolute_path?(value)
-        fail Puppet::Error, "File paths must be fully qualified, not '#{value}'"
+        fail Puppet::Error, "File paths must be fully qualified"
```

Comment:
Huh, I wonder why value didn't expand when I was testing earlier, perhaps
something about my environment? I rolled back this change.

On Thu, Jul 21, 2016 at 12:55 PM, Kylo Ginsberg <notifications@github.com>
wrote:

> In lib/puppet/type/file.rb
> <https://github.com/puppetlabs/puppet/pull/5126#discussion_r71775409>:
>
> > @@ -51,7 +51,7 @@ def self.title_patterns
> >
> >      validate do |value|
> >        unless Puppet::Util.absolute_path?(value)
> > -        fail Puppet::Error, "File paths must be fully qualified, not '#{value}'"
> > +        fail Puppet::Error, "File paths must be fully qualified"
>
> It looks like it expands in the snippet below:
>
> > be puppet apply -e 'file { "foo": }'
> Notice: /File[/Users/kylo/.puppetlabs/etc/code/environments/production]/ensure: created
> Notice: Compiled catalog for kylo.corp.puppetlabs.net in environment production in 0.04 seconds
> Error: Parameter path failed on File[foo]: File paths must be fully qualified, not 'foo' at line 1
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/puppetlabs/puppet/pull/5126/files/883ba1c7aa3a0935d40f984175dd76b0f1352374#r71775409>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AMVyKgYvGaeyOyc5cP1433_v-jx538RDks5qX87AgaJpZM4JQCqN>
> .
>



-- 
*Andy Henroid*
Senior Software Engineer
andy.henroid@puppet.com